request.go

```diff
@@ -6,12 +6,13 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"path"
 	"strings"
 
 	"github.com/aws/aws-lambda-go/events"
 )
 
-func newHTTPRequest(ctx context.Context, event events.APIGatewayProxyRequest) (*http.Request, error) {
+func newHTTPRequest(ctx context.Context, event events.APIGatewayProxyRequest, useProxyPath bool) (*http.Request, error) {
```

Comment:
consider passing in `*Options` so the method is positioned for any other options that may need to be passed through.

Reply:
Initially I considered passing `*Options`, but then I decided to pass only `useProxyPath` value because `newHTTPRequest` needs to know only about that particular option.